### ISSUE TITLE
Generated Data for Augmentation

### ISSUE THREAD

**Issue by mrabiabrn** [cure-lab/MagicDrive#93]
Hi,

I noticed the code for validation set generations but didn't find any code for training data augmentation. Should we follow the same procedure for generating training data? Could you provide more details on this or share a link to the generated data for BEV perception? That would be greatly appreciated.

**Comment by flymin** [comment]
Yes, you can refer to the procedure for validation set. They are similar.

**Comment by mrabiabrn** [comment]
Hi,
I tried to generate new samples from training data using the provided validation set generation script. However, I realized that for training instances, generations are not diverse and quite similar to the original data (color of the vehicles, shape of the road, background, etc.) This is not the case for validation samples. I can see diverse generations for the same bounding boxes. I added the training and validation generation results below. What do you think could be the reason for this?                                     
                                                 
![n015-2018-07-11-11-54-16+0800__CAM_BACK__1531281439787525ori](https://github.com/user-attachments/assets/c87320a2-4b10-4060-b91d-424c2afe3ad6)

**Comment by mrabiabrn** [comment]
Hi,
I tried to generate new samples from training data using the provided validation set generation script. However, I realized that for training instances, generations are not diverse and quite similar to the original data (color of the vehicles, shape of the road, background, etc.) This is not the case for validation samples. I can see diverse generations for the same bounding boxes. I added the training and validation generation results below. What do you think could be the reason for this?

Validation original vs generated
![image](https://github.com/user-attachments/assets/12f24e55-faf9-4868-98d6-433e56fc3841)


Training original vs generated
![image](https://github.com/user-attachments/assets/b6519f15-1a1d-4e43-bff9-16a14d780612)

**Comment by flymin** [comment]
In some cases, this may happen. However, using such data to augment the original training set leads to improvements in downstream tasks.

If it is severe in your case, you can try editing the scene condition to generate different data for augmentation.

**Comment by mrabiabrn** [comment]
Training generations are generally like this in my case. It doesn't improve CVT performance, it even hurts it. I can try editing the scene and text condition, but to reproduce your results, it would be great if you could share your training and validation set generations so I can identify any discrepancies and get diverse generations.

**Comment by flymin** [comment]
We already released the model weights. You can sample our model and see it.

I cannot share the data. However, I have to admit that our cases on the training set are similar to yours. We did not modify the code for perception models; we only added more generated data as described in our paper. Maybe you can also try bevfusion and see.

**Comment by RYHSmmc** [comment]
@flymin Hello author, the road segmentation performance of CVT in Table 1 is 61, which is confirmed to be 59.3 in Table 4. We also reproduced 59.x. How was 61 obtained? Or what is the difference between these two data?

**Comment by flymin** [comment]
> which is confirmed to be 59.3 in Table 4

This is not true. Please also see Figure 7. I think the problem lies in $M=\{0\}$.

**Comment by mrabiabrn** [comment]
To confirm, the results in Table 1 were generated with use_zero_map_as_unconditional = True and a guidance scale of 2. Is this correct?

**Comment by flymin** [comment]
Yes. And sorry for the late reply.
